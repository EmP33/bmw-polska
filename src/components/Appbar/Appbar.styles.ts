import styled from "styled-components";

export const Container = styled.nav`
  width: 100%;
  color: var(--color-white);
  align-self: start;
  display: grid;
  grid-template-columns: 1fr max-content;
  margin-top: 1rem;
  column-gap: 2rem;
  align-items: center;

  svg {
    width: 40px;
    height: 40px;
  }

  ul {
    list-style-type: none;
    display: flex;
    column-gap: 3rem;
    border-bottom: 1px solid rgba(255, 255, 255, 0.25);
    padding-top: 0.5rem;

    @media only screen and (max-width: 900px) {
      display: none;
    }

    li {
      font-size: 1.125rem;
      font-family: var(--font-heading);
      font-weight: bold;
      letter-spacing: 0.9px;
      padding-bottom: 1rem;
      border-bottom: 3px solid transparent;
      cursor: pointer;

      &:hover {
        border-bottom: 3px solid var(--color-blue);
      }
    }
  }

  p {
    font-size: 1rem;
    font-family: var(--font-heading);
    font-weight: 400;
    justify-self: end;
    padding-top: 0.5rem;

    @media only screen and (max-width: 900px) {
      display: none;
    }

    span {
      font-weight: bold;
    }
  }
`;
