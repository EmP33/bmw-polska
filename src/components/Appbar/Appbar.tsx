import React from "react";
// Styles
import { Container } from "./Appbar.styles";
// Images
import { BmwLogo } from "../../assets";
import { HamburgerButton } from "../../Components.styles";

interface Props {
  toggleMenu: () => void;
}

const Appbar: React.FC<Props> = ({ toggleMenu }) => {
  return (
    <Container>
      <HamburgerButton onClick={toggleMenu}>
        <span></span>
        <span></span>
        <span></span>
      </HamburgerButton>
      <ul>
        <li>Nowe BMW iX</li>
        <li>Umów jazdę próbną</li>
      </ul>
      <BmwLogo />
      <p>
        <span>Radość</span> z jazdy
      </p>
    </Container>
  );
};

export default Appbar;
