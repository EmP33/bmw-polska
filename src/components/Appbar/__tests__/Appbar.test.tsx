import Appbar from "../Appbar";
import ReactShallowRenderer from "react-test-renderer/shallow";

describe("<Appbar />", () => {
  it("should render properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Appbar />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
