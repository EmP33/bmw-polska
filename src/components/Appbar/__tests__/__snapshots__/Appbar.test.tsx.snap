// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Appbar /> should render properly 1`] = `
<styled.nav>
  <styled.button>
    <span />
    <span />
    <span />
  </styled.button>
  <ul>
    <li>
      Nowe BMW iX
    </li>
    <li>
      Umów jazdę próbną
    </li>
  </ul>
  <BmwLogo />
  <p>
    <span>
      Radość
    </span>
     z jazdy
  </p>
</styled.nav>
`;
